app/geo/gui/src/ts/twxplore/gui/geo/states/map/FeaturesByType.ts

```diff
@@ -1,10 +1,10 @@
 import {MapFeature} from "./MapFeature";
 import {MapFeatureTypeState} from "./MapFeatureTypeState";
-import {FilterStateTypes} from "./MapFilterStateTypes";
+import {MapFeatureAttributeState} from "./MapFeatureAttributeState";
 
 export interface FeaturesByType {
   features: MapFeature[];
   dirty: boolean;
   featureTypeState: MapFeatureTypeState;
-  attributesState: {[attribute: string]: FilterStateTypes};
+  attributeStates: {[featureAttributeName: string]: MapFeatureAttributeState};
```

Comment:
I'm not sure TS allows this, but perhaps this could be [keyof MapFeature]

Reply:
Didn't allow 'Union-types' to be used as an index unfortunately